### ISSUE TITLE
[question]: How to Capture silent notification data

### ISSUE THREAD

**Issue by sunilsinghchaudhary00** [OneSignal/OneSignal-Flutter-SDK#985]
### How can we help?

I am trying to capture the silent notification data but unable to do that showing data in log but its onesignal data not my logs 
i am getting silent notif data in backgorund onesignal logs 

D/OneSignal(28679): [DefaultDispatcher-worker-1] Saving Notification id=d4eed-9291-4a21-aa66-bf11fb6e5eba
D/OneSignal(28679): [DefaultDispatcher-worker-1] Notification saved values: full_data={"google.delivered_priority":"normal","google.sent_time":1735539186317,"google.ttl":2419200,"google.original_priority":"normal","custom":"{\"i\":\"d4eed-9291-4a21-6-bf11fb6e5eba\",\"a\":{\"body\":\"{\\\"description\\\":\\\"Testing Silent Notification\\\"}\",\"click_action\":\"FLUTTER_NOTIFICATION_CLICK\",\"description\":\"Testing Silent Notification\",\"title\":\"Testing Silent Notification\"}}","google.product_id":11181503,"from":"38108206621","google.message_id":"0:173553915688%00ae9cb1f9fd7ecd","google.c.sender.id":"381086206621"}

my code is 

   void initonesignal() {
    OneSignal.Debug.setLogLevel(OSLogLevel.verbose);
    OneSignal.Debug.setAlertLevel(OSLogLevel.none);
    OneSignal.consentRequired(false);
    OneSignal.initialize("40a574c6-b53f-4277-98bb-e42e96a6028f");
    OneSignal.User.pushSubscription.addObserver((state) async {
      global.onesignalId = OneSignal.User.pushSubscription.id;
      print("Opted Push Subscription Id ${global.onesignalId}");
      print("Opt Subscription Token ${OneSignal.User.pushSubscription.token}");
      await saveOneSignalId(global.onesignalId!);
    });

    OneSignal.Notifications.addPermissionObserver((state) {
      log("Has permission $state");
    });

    OneSignal.Notifications.addClickListener((event) {
      try {
        onSelectNotification(event.notification.additionalData);
      } catch (e) {
        log('Failed to decode handleNotificationData: $e');
      }
    });
    OneSignal.Notifications.addForegroundWillDisplayListener((event) {
      event.preventDefault();
      event.notification.display();
      log('capture onesignal ${event.notification.additionalData} ');
      showForegroundNotification(event.notification.additionalData);
    });
  }

custom sound working finee
normal default sound working fine capture data in logs fine
but in silent notif no data is capture in logs pls help anybody have any idea how to do that really helpfull

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by diego-aquila** [comment]
@sunilsinghchaudhary00 have you managed to solve this question? How did you do it?

**Comment by sunilsinghchaudhary00** [comment]
yes i found a workaround the thing is you have to do code on android mainactivity which tidious and also its not good idea using like that because it is executing many times so good way to do this is get the silent data using firebase not with onesignal as onesignal is for only notification focused use Firebase for that

**Comment by fadi-george** [comment]
Thanks for reaching out, and sorry for the delayed response.

`addForegroundWillDisplayListener` only runs while the app is in the foreground, and `addClickListener` runs when the user opens a notification. The Flutter SDK does not currently provide a background notification-received callback or wake a Dart isolate for silent notifications.

For silent data handling on Android, please use a native OneSignal Notification Service Extension: https://documentation.onesignal.com/docs/en/service-extensions. You can process `notification.additionalData` there and persist any state for Flutter to read when the app opens. Similar questions are covered in #913 and #1018.

Since this is expected behavior and the supported approach is native handling, I’ll close this issue. Thank you!
